htdocs/don/payment/payment.php

```diff
@@ -192,14 +192,14 @@
 	print '</tr>';
 
 	print '<tr><td class="fieldrequired">'.$langs->trans("PaymentMode").'</td><td colspan="2">';
-	$form->select_types_paiements(GETPOSTISSET("paymenttype") ? GETPOST("paymenttype") : $object->paymenttype, "paymenttype");
+	$form->select_types_paiements(GETPOSTISSET("paymenttype") ? GETPOST("paymenttype") : $object->fk_typepayment, "paymenttype");
 	print "</td>\n";
 	print '</tr>';
 
 	print '<tr>';
 	print '<td class="fieldrequired">'.$langs->trans('AccountToCredit').'</td>';
 	print '<td colspan="2">';
-	$form->select_comptes(GETPOSTISSET("accountid") ? GETPOST("accountid") : $object->accountid, "accountid", 0, '', 2); // Show open bank account list
+	$form->select_comptes(GETPOSTISSET("accountid") ? GETPOST("accountid") : $object->fk_account, "accountid", 0, '', 2); // Show open bank account list
```

Comment:
$object seems to be the class new Don()
For this class, there is no bank account into database, into the fetch and into screen to create and edit.
So i think the fix is different and we should replace the $object->accountid with "0"  (there is no default bank account stored into a donation that we can reuse here as a default value for the combobox).

Reply:
Ok i see i thought the first param of the function should be required and contain a value, now i understand it can have also a default value but why "0" in place " " ?

Reply:
Yes, having 0 instead of $object->accountid that is undef seems the good way to fix this case.

Reply:
ok i will fix it